### DISCUSSION TITLE
HiDPI-Mode does not work correctly on OS8

### DISCUSSION THREAD

**Discussion by NutriPhil** [elementary/triage#617] [Q&A]
Hi, i encountered a problem with two different high resolution monitors. One is the internal screen of a Macbook Pro, the other one beeing a 32" Ultra HD / 4K external Monitor. On both screens the HighDPI-Mode (2 and 3) wont work correct. While the normal mode makes everything to small, HighDPI 2 is good for me and may resemble best the 150% mode under standard Ubuntu. The menu bar looks quiet good as well, for my opinion its a little bit too big and a semi transparent mode would be nice, but the real problem is, that all windows i expand seem to try to align themselves to the bar of the standard resolution mode, which is half the thickness. So the title-bar of the windows get half way under the High DPI-title bar. To grab the window there is only a small part left, and it looks strange as well. Besides i would really like to see the functionality of the "Pantheon tweaks" implemented or preinstalled to the system, not only because of the themeability but also the possibility to alter window controls more to a Mac or Windows-like experience.

**Comment by teamcons** [comment]
Hello

Pantheon Tweaks is on Flathub. It works pretty nice ! On 8, Flathub is preinstalled. On 7, the flathub Installation page explains how to add it - but it is pretty much one or two commands away.
You can also download the flatpakref file from flathub, and Sideload will take care of adding it for you :)

Alternatively to display sizes, have you tried playing with font size ? In eOS it changes the UI size as well
It is what we use for the box on the TV, to avoid messing with DPI

Could you send a screenshot of the title-bar of the windows half way under the HDPI-title bar, along with more info about your hardware ? Is it eOS 7 or 8 ? I need to check if there is an open issue or not, this sound like a bug
As a workaround, i believe there is a shortcut to move/resize windows (Alt+F8 to resize, but i dont remember the Move one)

Point taken for the semi transparent mode. I'll look into it if i have time.

**Comment by NutriPhil** [comment]
Im testing the RC of elementary OS 8 from 13th of november 2024. All updates from the update center applied today.

Normal Resolution (a bit to small for my opinion on 4k 32") but everything works just normal, despite the dock-Icons jumping out of the dock when clicked. I enjoy the Apple-like effect, but the icons got cut of at the top while jumping. So it does not look exactly good. Right-click works fine. Windows allign perfectly to the top bar.

Video proof:

- no longer available -

In HighDPI2 (and 3) its different. Icons have the same problem btw. Windows allign half way under (!) the actual Top-Menu-Bar, and to me it looks like they allign to the size of the Normal Resolution bar. Right click works, but the context menu appears far away from the mouse cursor. So this is an issue as well. Right click in the dock works normal.

Video proof:

- no longer available -

**Comment by NutriPhil** [comment]
Hardware configuration under my current Zorin OS daily driver:

![Bildschirmfoto vom 2024-11-20 15-40-39](https://github.com/user-attachments/assets/09f04dc3-b6fc-47f0-b1f2-d3d9c9683e41)

**Comment by teamcons** [comment]
The icon clipping in the dock is known and reported already: https://github.com/elementary/dock/issues/275 and discussed in a PR here https://github.com/elementary/dock/pull/307

Btw you can grab the window from the top bar when maximized. that avoids you to snipe

The HIDPI one im not sure . I assume you are using the X11/"Classic" session right ?
I have reported it here: https://github.com/elementary/triage/issues/618

**Comment by NutriPhil** [comment]
I did Not alter the Session preferences , so i use the "Standard" i guess. Ist It already Wayland?

Sorry, i forgot to mention, that changing font size does not solve the problem.

What about the right click context menu? Has this already been reported? 

I have the same issues on the PC i gave you the Hardware Specs as well as with a 13" MBP with "Retina-Display".

**Reply by teamcons** [reply]
No, thats X11.

I dont know. I dont have the time to go through Issues on github. Feel free to take a look
But it may be a similar issue with scaling

**Comment by NutriPhil** [comment]
Ill take a look. Thank you! :-)

**Reply by trevorsauerbrey** [reply]
Did you ever get this resolved? I'm experiencing the same overlap and it is frustrating. Thanks!

**Reply by syneryder** [reply]
The issue with the "Top Menu Bar" (actually called Wingpanel) overlapping the top half of windows in OS8 has finally been fixed (as of 13 March 2025). Try checking for updates in System Preferences. It was a bug in the Wingpanel code that ignored scaling, see this issue comment here:

https://github.com/elementary/wingpanel/issues/594#issuecomment-2704313080

and resolved here:

https://github.com/elementary/wingpanel/pull/601

I'm still not seeing a fix for the wildly incorrect positioning of right-click context menu on the desktop, though, that looks like another dual-monitor and scaling issue.
